src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,126 @@
+package duke.tasklist;
+
+import duke.tasks.Task;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.ToDo;
+
+import duke.exception.DukeException;
+
+import duke.tasks.Task;
+
+import java.util.ArrayList;
+
+import java.util.List;
+
+/**
+ * Represents the list of tasks in the chatbot.
+ */
+public class TaskList {
+
+    ArrayList<Task> taskList;
+
+    /**
+     * Constructor for the taskList object.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Marks the specified task as being done.
+     *
+     * @param number Number of the task, starting from index 1.
+     * @return The task being marked as done.
```

Comment:
Should the exception thrown be explictly mentioned here?

```suggestion
     * @param number Number of the task, starting from index 1.
     * @throws DukeException if the user inputs an index that is out of bounds
     * @return The task being marked as done.
```